elastic/kibana
# [Response Ops][Alerting] Add limit to `size` when ES query rule specifies a group by field

Towards https://github.com/elastic/kibana/issues/188466

## Summary

When retrieving hits for an ES query rule with a group by field, we use the top hits aggregation. This aggregation limits the number of returned hits to 100 unless the index level setting `index.max_inner_result_window` is changed. This PR updates the UI to add validation that size is <= 100 when a group by field is specified and updates the rule executor to cap the top hits size. This allows existing rules with this undesired configuration to continue running.

## To Verify

1. Run `main` and create an ES query rule, both KQL and DSL. Add a group by field and then set the size to 300.
2. Let the rules run and see that they fail with an error
3. Update to this branch and see the rule executions start to succeed with a warning about capping the top hits size get logged. Edit the rules in the UI and see a new validation error that prevents you from saving the rule without reducing the size.
